action.yml

```diff
@@ -23,26 +23,26 @@ inputs:
     description: 'The path to repo when code is analyzed with clang-tidy; may set to "/__w" for users who run clang-tidy in a docker container'
     required: false
     default: '/home/runner/work'
-
-
 runs:
   using: 'composite'
   steps:
     - name: Setup Python
       uses: actions/setup-python@v5
       id: setup-python
       with:
-        python-version: "3.11"
+        python-version: 3.11
         update-environment: false
     - name: Setup venv
-      run: ${{ steps.setup-python.outputs.python-path }} -m venv ${GITHUB_ACTION_PATH}/venv
-
+      run: |
+        "${{ steps.setup-python.outputs.python-path }}" -m venv "${GITHUB_ACTION_PATH}/venv"
       shell: bash -l {0}
```

Comment:
I'm not sure why exactly we run `bash` in the "login shell mode" here and in other similar places. Do we really need it to read and execute various profiles, `~/.bash_login`, `~/.bash_logout`? It has something to do with the launch on a self-hosted runners? Hi @renefritze could you please comment on this?

Reply:
AFAIK that's what actions would use as a default for a linux run step when not specifying a shell.

Reply:
@renefritze According to [manual](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsshell) the [default](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell) is `bash -e {0}` and if we set just `bash` here, the much stricter version `bash --noprofile --norc -eo pipefail {0}` will be used. Will there be any problems in the self-hosted environment due to a stricter shell configuration?

Reply:
Ah, alright. No problems with self-hosted I'm aware of, but I'm not an extensive user.